pipeline/subtraction.py

```diff
@@ -225,49 +236,335 @@ def _subtract_zogy(self,
+            #   throwing away all the stars in (say) one corner of the
+            #   image.
+            # This is going to crash if there isn't a FWHM_IMAGE
+            #   field in ref_sources and new_sources.  If that happens,
+            #   hopefully you came here and saw this comment.  Those
+            #   fields really should be there if sources were extracted
+            #   with sextractor, and right now there are lots of things
+            #   in the code that depend on stuff sextractor does that
+            #   sep doesn't.
+            width_ratio = new_sources.data['FWHM_IMAGE'] / new_psf.fwhm_pixels
+            wgood = ( width_ratio > 0.9 ) & ( width_ratio < 1.1 ) & ( new_sources.good )
+
+            newstars = pandas.DataFrame( { 'x': new_sources.x[wgood],
+                                           'y': new_sources.y[wgood],
+                                           'flux': new_sources.psffluxadu()[0][wgood] } )
+            newsc = new_wcs.wcs.pixel_to_world( newstars.x, newstars.y )
+
+            # wgood = ref_sources.is_star & ref_sources.good
+            width_ratio = ref_sources.data['FWHM_IMAGE'] / ref_psf.fwhm_pixels
+            wgood = ( width_ratio > 0.9 ) & ( width_ratio < 1.1 ) & ( ref_sources.good )
+            refstars = pandas.DataFrame( { 'x': ref_sources.x[wgood],
+                                           'y': ref_sources.y[wgood] } )
+            refsc = ref_wcs.wcs.pixel_to_world( refstars.x, refstars.y )
+
+            matchdex, sep2d, _ = astropy.coordinates.match_coordinates_sky( refsc, newsc )
+            wmatch = np.where( sep2d < 1 * units.arcsecond )[0]
+
+            refstars = refstars.iloc[ wmatch ].reset_index()
+            newstars = newstars.iloc[ matchdex[wmatch] ].reset_index()
+
+            SCLogger.debug( f"_subtract_hotpants matched {len(newstars)} stars between new and ref" )
+
+            # How many stars do we want?  Imagine dividing the image
+            # into regions of size 256; we want at least one star per
+            # region.  (This is kinda arbitrary, but whatevs.)
+            nstars_ideal = ( new_image.data.shape[0] // 256 ) * ( new_image.data.shape[1] // 256 )
+
+            if len(newstars) < nstars_ideal // 8:
+                SCLogger.error( f"_subtract_hotpants: {len(newstars)} stars for PSF matching isn't enough" )
+                raise RuntimeError( "Not enough stars for _subtract_hotpants PSF matching" )
+            if len(newstars) < nstars_ideal // 4:
+                SCLogger.warning( f"_subtract_hotpants: {len(newstars)} stars for PSF matching is low, "
+                                  f"reducing kernel orders" )
+                hotgaussorders = [ 1, 0 ]
+                hotgaussfactors = [ 0.6, 1.2 ]
+            if len(newstars) > nstars_ideal * 2:
+                # embarssment of riches, crop the list to make hotpants run faster
+                sortdex = numpy.argsort( newstars.flux )
```

Comment:
```suggestion
                sortdex = np.argsort( newstars.flux )
```

Reply:
Unless you want to change the import statement instead

Reply:
I do want to, but also don't think we should.

(I much prefer using `numpy.` everywhere-- I have been bitten by namespace pollution before, and on the balance between "distinct and explicit" and "too much to type", I usually lean towards the former, *especially* when we're just talking 3 extra letters.  But the standard that everybody else uses is `np.`, and since this pipeline isn't my private playground, we should stick with what most people do.)